From d46f77108a85453791ad0dd8a4ae0ec6fb6dedb5 Mon Sep 17 00:00:00 2001
From: Dmitri Epshtein <dima@marvell.com>
Date: Mon, 21 Oct 2013 16:24:49 -0400
Subject: [PATCH 1041/1825] fix: pp2: set MaximumLowLatencyPacket to 256 bytes

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit f1832c3584cfd3f24ed3d012e0891ae85a299440

	Define fields of RX Control register (offset: 0x140)
	Add setting default value to RX control register

Change-Id: I395af8a5ec8b7044b04f4dd7f84d0c9de3bcc205
Signed-off-by: Dmitri Epshtein <dima@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/3801
Reviewed-by: Uri Eliyahu <uriel@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 arch/arm/plat-armada/mv_hal/pp2/gbe/mvPp2Gbe.c     |    3 +++
 arch/arm/plat-armada/mv_hal/pp2/gbe/mvPp2GbeRegs.h |   16 ++++++++++++++++
 2 files changed, 19 insertions(+), 0 deletions(-)

diff --git a/arch/arm/plat-armada/mv_hal/pp2/gbe/mvPp2Gbe.c b/arch/arm/plat-armada/mv_hal/pp2/gbe/mvPp2Gbe.c
index 90babe0..71fe404 100644
--- a/arch/arm/plat-armada/mv_hal/pp2/gbe/mvPp2Gbe.c
+++ b/arch/arm/plat-armada/mv_hal/pp2/gbe/mvPp2Gbe.c
@@ -249,6 +249,9 @@ MV_STATUS mvPp2DefaultsSet(int port)
 		mvPp2WrReg(MV_PP2_TXP_SCHED_PERIOD_REG, mvPp2HalData.tClk / 1000000);
 		mvPp2TxpMaxRateSet(port, txp);
 	}
+	/* Set MaximumLowLatencyPacketSize value to 256 */
+	mvPp2WrReg(MV_PP2_RX_CTRL_REG(port), MV_PP2_RX_USE_PSEUDO_FOR_CSUM_MASK |
+						MV_PP2_RX_LOW_LATENCY_PKT_SIZE_MASK(256));
 
 	/* Enable Rx cache snoop */
 	if (mvPp2HalData.iocc)
diff --git a/arch/arm/plat-armada/mv_hal/pp2/gbe/mvPp2GbeRegs.h b/arch/arm/plat-armada/mv_hal/pp2/gbe/mvPp2GbeRegs.h
index a7fe021..4ddfe52 100644
--- a/arch/arm/plat-armada/mv_hal/pp2/gbe/mvPp2GbeRegs.h
+++ b/arch/arm/plat-armada/mv_hal/pp2/gbe/mvPp2GbeRegs.h
@@ -138,6 +138,22 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #define MV_PP2_RX_CTRL_REG(port)		(MV_PP2_REG_BASE + 0x140 + 4 * (port))
 
+#define MV_PP2_RX_GEMPID_SRC_OFFS		8
+#define MV_PP2_RX_GEMPID_SRC_MASK		(0x7 << MV_PP2_RX_GEMPID_SRC_OFFS)
+
+#define MV_PP2_RX_LOW_LATENCY_PKT_SIZE_OFFS	16
+#define MV_PP2_RX_LOW_LATENCY_PKT_SIZE_BITS	12
+#define MV_PP2_RX_LOW_LATENCY_PKT_SIZE_MAX	((1 << MV_PP2_RX_LOW_LATENCY_PKT_SIZE_BITS) - 1)
+#define MV_PP2_RX_LOW_LATENCY_PKT_SIZE_MASK(s)	(((s) & MV_PP2_RX_LOW_LATENCY_PKT_SIZE_MAX) << \
+							MV_PP2_RX_LOW_LATENCY_PKT_SIZE_OFFS)
+
+#define MV_PP2_RX_DROP_ON_CSUM_ERR_BIT		30
+#define MV_PP2_RX_DROP_ON_CSUM_ERR_MASK		(1 << MV_PP2_RX_DROP_ON_CSUM_ERR_BIT)
+
+#define MV_PP2_RX_USE_PSEUDO_FOR_CSUM_BIT	31
+#define MV_PP2_RX_USE_PSEUDO_FOR_CSUM_MASK      (1 << MV_PP2_RX_USE_PSEUDO_FOR_CSUM_BIT)
+/*-------------------------------------------------------------------------------*/
+
 /************************** Descriptor Manager Top Registers ******************************/
 
 #define MV_PP2_RXQ_NUM_REG			(MV_PP2_REG_BASE + 0x2040)
-- 
1.7.5.4

